mcs/class/corlib/System.IO/DirectoryInfo.cs

```diff
@@ -424,39 +424,51 @@ public IEnumerable<FileSystemInfo> EnumerateFileSystemInfos (string searchPatter
 			return EnumerateFileSystemInfos (FullPath, searchPattern, searchOption);
 		}
 
-		static internal IEnumerable<FileSystemInfo> EnumerateFileSystemInfos (string full, string searchPattern, SearchOption searchOption)
+		static internal IEnumerable<FileSystemInfo> EnumerateFileSystemInfos (string basePath, string searchPattern, SearchOption searchOption)
 		{
-			string path_with_pattern = Path.Combine (full, searchPattern);
-			IntPtr handle = IntPtr.Zero;
-			MonoIOError error;
-			FileAttributes rattr;
-			bool subdirs = searchOption == SearchOption.AllDirectories;
+			Path.Validate (basePath);
+
+			IntPtr findHandle = IntPtr.Zero;
 
-			Path.Validate (full);
-			
 			try {
-				string s = MonoIO.FindFirst (full, path_with_pattern, out rattr, out error, out handle);
-				if (s == null)
-					yield break;
-				if (error != 0)
-					throw MonoIO.GetException (Path.GetDirectoryName (path_with_pattern), (MonoIOError) error);
+				string filePath;
+				int attrs;
+
+				try {} finally {
+					string basePathWithPattern = Path.Combine (basePath, searchPattern);
+
+					findHandle = MonoIO.FindFirstFile (basePathWithPattern, out filePath, out attrs, out int error);
+					if (((MonoIOError) error) != MonoIOError.ERROR_SUCCESS
+						 && ((MonoIOError) error) != MonoIOError.ERROR_FILE_NOT_FOUND)
+					{
+						throw MonoIO.GetException (Path.GetDirectoryName (basePathWithPattern), (MonoIOError) error);
+					}
+				}
 
 				do {
-					if (((rattr & FileAttributes.ReparsePoint) == 0)){
-						if ((rattr & FileAttributes.Directory) != 0)
-							yield return new DirectoryInfo (s);
+					if (filePath == null)
+						yield break;
+
+					if (filePath == "." || filePath == "..")
+						continue;
+
+					string fullPath = Path.Combine (basePath, filePath);
+
+					if (((((FileAttributes) attrs) & FileAttributes.ReparsePoint) == 0)) {
+						if ((((FileAttributes) attrs) & FileAttributes.Directory) != 0)
+							yield return new DirectoryInfo (fullPath);
 						else
-							yield return new FileInfo (s);
+							yield return new FileInfo (fullPath);
 					}
 
-					if (((rattr & FileAttributes.Directory) != 0) && subdirs)
-						foreach (FileSystemInfo child in EnumerateFileSystemInfos (s, searchPattern, searchOption))
+					if (((((FileAttributes) attrs) & FileAttributes.Directory) != 0) && searchOption == SearchOption.AllDirectories)
+						foreach (FileSystemInfo child in EnumerateFileSystemInfos (fullPath, searchPattern, searchOption))
 							yield return child;
 
-				} while ((s = MonoIO.FindNext (handle, out rattr, out error)) != null);
+				} while (MonoIO.FindNextFile (findHandle, out filePath, out attrs, out int _));
```

Comment:
You could even use `out _` for the last one

Reply:
I tried and the `Linux x64 mcs` status check failed because of "Unknown variable" or an equivalent message.

Reply:
`System.IO/DirectoryInfo.cs(468,76): error CS0103: The name `_' does not exist in the current context` at https://jenkins.mono-project.com/job/test-mono-pull-request-amd64-mcs/2259/parsed_console/log_content.html#ERROR1